packages/flutter_tools/bin/macos_assemble.sh

```diff
@@ -3,244 +3,32 @@
-  if [[ $ACTION == "clean" ]]; then
-    exit 0
-  fi
-  BuildApp "prepare"
-}
-
-# Adds the App.framework as an embedded binary, the flutter_assets as
-# resources, and the native assets.
-EmbedFrameworks() {
-  # Embed App.framework from Flutter into the app (after creating the Frameworks directory
-  # if it doesn't already exist).
-  local xcode_frameworks_dir="${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}"
-  RunCommand mkdir -p -- "${xcode_frameworks_dir}"
-  RunCommand rsync -av --delete --filter "- .DS_Store" "${BUILT_PRODUCTS_DIR}/App.framework" "${xcode_frameworks_dir}"
-
-  # Embed the actual FlutterMacOS.framework that the Flutter app expects to run against,
-  # which could be a local build or an arch/type specific build.
-
-  # Copy Xcode behavior and don't copy over headers or modules.
-  RunCommand rsync -av --delete --filter "- .DS_Store" --filter "- Headers" --filter "- Modules" "${BUILT_PRODUCTS_DIR}/FlutterMacOS.framework" "${xcode_frameworks_dir}/"
-
-  # Sign the binaries we moved.
-  if [[ -n "${EXPANDED_CODE_SIGN_IDENTITY:-}" ]]; then
-    RunCommand codesign --force --verbose --sign "${EXPANDED_CODE_SIGN_IDENTITY}" -- "${xcode_frameworks_dir}/App.framework/App"
-    RunCommand codesign --force --verbose --sign "${EXPANDED_CODE_SIGN_IDENTITY}" -- "${xcode_frameworks_dir}/FlutterMacOS.framework/FlutterMacOS"
-  fi
-
-  # Copy the native assets. These do not have to be codesigned here because,
-  # they are already codesigned in buildNativeAssetsMacOS.
-  local project_path="${SOURCE_ROOT}/.."
-  if [[ -n "$FLUTTER_APPLICATION_PATH" ]]; then
-      project_path="${FLUTTER_APPLICATION_PATH}"
-  fi
-  local native_assets_path="${project_path}/${FLUTTER_BUILD_DIR}/native_assets/macos/"
-  if [[ -d "$native_assets_path" ]]; then
-    RunCommand rsync -av --filter "- .DS_Store" --filter "- native_assets.yaml" --filter "- native_assets.json" "${native_assets_path}" "${xcode_frameworks_dir}"
-
-    # Iterate through all .frameworks in native assets directory.
-    for native_asset in "${native_assets_path}"*.framework; do
-      [ -e "$native_asset" ] || continue # Skip when there are no matches.
-      # Codesign the framework inside the app bundle.
-      RunCommand codesign --force --verbose --sign "${EXPANDED_CODE_SIGN_IDENTITY}" -- "${xcode_frameworks_dir}/$(basename "$native_asset")"
-    done
-  fi
-}
+# exit on error, or usage of unset var
+set -euo pipefail
+
+# Needed because if it is set, cd may print the path it changed to.
+unset CDPATH
+
+function follow_links() (
+  cd -P "$(dirname -- "$1")"
+  file="$PWD/$(basename -- "$1")"
+  while [[ -h "$file" ]]; do
+    cd -P "$(dirname -- "$file")"
+    file="$(readlink -- "$file")"
+    cd -P "$(dirname -- "$file")"
+    file="$PWD/$(basename -- "$file")"
+  done
+  echo "$file"
+)
+
+PROG_NAME="$(follow_links "${BASH_SOURCE[0]}")"
+BIN_DIR="$(cd "${PROG_NAME%/*}" ; pwd -P)"
+FLUTTER_ROOT="$BIN_DIR/../../.."
+DART="$FLUTTER_ROOT/bin/dart"
```

Comment:
This now mimics what's done in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/bin/xcode_backend.sh (which iOS uses).